### ISSUE TITLE
Vitrum touchpanel only updates when I check the device manually

### ISSUE THREAD

**Issue by novaflash** [athombv/homey#527]
**What is your Homey version (Settings → System)?**

0.8.30

**What did you try to achieve?**
I have the Homey and a Vitrum touchpanel. It's a 3 button touchpanel, called the Vitrum Satellite EU III. When I push any of the three buttons, I want Homey to do some sort of action. I pushed the first button, so it would be in the 'on' state. I had a flow that was set to say a word the moment that device is switched 'on'.

**What was the result?**
Nothing happened until I went to Zones and Devices, and looked up the device, and clicked the slidey-bars-icon button. The moment I did that, it retrieved the current state of that button (correctly) and the flow then triggered.

**What did you expect as result?**
That I don't have to manually check what the state of the button is, and that the Homey would automatically see when I press the button on my Vitrum switch.

**Any other remarks**
It's a multi-channel devices, and according to the manual, it has an association group for each button (groups 1, 2 and 3, in this case). When I add the device to Homey, Homey asks which association group it reports back to the controller on. I tried all the possible settings (none, 1, 2, or 3). I also tried going to the settings of the main device and setting the association groups all to '1' (the Homey's Z-Wave ID and SIS ID).

There are no other Z-Wave devices connected to Homey, this is the only one.

**Comment by novaflash** [comment]
ps; I'd be willing to loan you guys one if it would help you fix this.

**Comment by ocatie** [comment]
Sound like the device isn't sending updates to Homey. When you open the device card the state is fetched and thus updated if changed. Do you know what command class the device uses to notify button presses?

If you could send a log and tell the node id of the device, I'll take a look to see if any data is being received.

**Comment by novaflash** [comment]
Hi. Thank you for paying attention to this issue. I'll get right on it.

My Homey has Z-Wave ID 1, SIS 1, and the Vitrum switch has node ID 2. There are 3 slave nodes, one for each button. Those are 2.1, 2.2, and 2.3. There are no other nodes.

When I looked in the manual for this device, I can find no mention of a command class. Only that it has as many association groups as there are buttons. So for a 3 button switch, there are 3 associations group. And the maximum nodes that can be defined for each association is, according to the manual, 5. With a maximum of 4 endpoints per association.

I currently have them all set to associate with node ID 1 (the Homey itself) so it should report its status there, in theory.

When I look at a device called the vitrum remote, I see some command class information. See here;
http://www.switchtovitrum.com/wp-content/uploads/2014/12/VITRUM-REMOTE-CONTROL_EN.pdf

And the manuals/documents for the exact device I have (Vitrum Satellite EU III):
http://www.switchtovitrum.com/schedetecniche/EN/01E030030.pdf (datasheet)
http://www.z-wave.ru/podderzhka/skachat/category/83-thinksimple.html?download=235:vitrum-satellite&start=20 (user manual that comes with the device)
http://www.vdagroup.com/Vitrum-home-control.pdf (sales document)

**LOG ID: 13E88DD4C8**

**Comment by novaflash** [comment]
I am currently on version 0.8.32 and the behavior hasn't changed. Would it make any sense for me to unpair this device, reset my Homey Z-wave chip, and try to pair it again, to start with a 'clean slate' with the latest firmware?

**Comment by ocatie** [comment]
The device is sending Basic frames. These are used to control other devices and aren't parsed by Homey.

**Comment by novaflash** [comment]
Why do I get the horrible feeling that this is never going to be parsed by the Homey, ever?

**Comment by ocatie** [comment]
Because these are frames to control devices. Homey is a controller (does the controlling) and doesn't have actuators exported to the Z-Wave network.

In the future we will support the Central Scene command class to trigger flows.

**Comment by novaflash** [comment]
I regret that I don't understand exactly what is going on. For me, these are switches that send commands, on/off, to the controller it is paired to. Why it doesn't work is completely beyond my understanding at this point.

Can I take your last line about the central scene command class as a solution that will make these switches work with Homey? So I just need to be patient and eventually this will work?

**Comment by ocatie** [comment]
If your device can be configured to send Central Scene command, yes definitely this will work in the future!

**Comment by novaflash** [comment]
What kind of an answer is that? It can't be configured to do a central scene command. There is nothing in the manual to do anything like that. It does what it does. It sends its signal to the controller. I know it does, you even confirmed it. It's standard Z-wave. If the Homey can't deal with this signal and doesn't want to work with it, then I obviously have the wrong product in my hands. Please reconsider closing this ticket and please make it work.

For your information, on my HC2, it _DOES_ work. Configured _EXACTLY_ the way it is now.

**Comment by ocatie** [comment]
I wish is was as simple as just supporting that command but Z-Wave is a closed protocol and we have to comply to a lot of rules.

If the device can't be configured to use the Central Scene command class an app can make it work regardless.

**Comment by novaflash** [comment]
Thank you for responding.

I am very relieved that this doesn't mean the end for compatibility between Vitrum and Homey! I assume though that this option of adding an app is something that cannot yet be done because Homey's Z-wave implementation isn't at the point yet where apps for specific Z-wave devices are possible? I suppose I'll have to wait. Then perhaps I'll be able to get some sample code and give it a crack.

**Comment by bvdbos** [comment]
Don't know much about Zwave but when using a controller with the basic-class we wouldn't have to use association-groups as these are used for communicating between devices directly... The manual states (chapter 3) the "include" function should be used, not the associate-function (chapter 4)? Did you use "INC" or "ASC"?

**Comment by novaflash** [comment]
Hello @bvdbos ,

I appreciate your input, so please allow me try to explain where you're going wrong. 

The include function is to add an as yet unpaired device to an existing Z-wave network. This is done via the controller. So, when I go to the Homey, and I go to Zones & Devices, and I click the + sign, the Homey is now waiting for a signal from an unpaired device, to be added to the Homey Z-wave network. That process works. I can see the devices just fine. I can even control the switches from the Homey; that part works fine. The Vitrum is listening to the Homey just fine and the Vitrum switch has been 'included' in the network just fine.

Association is the act of telling the Vitrum device to send its on/off signals to a specific device. According to the documentation it is possible to assign up to 5 devices to each button. This process is called association. This can even be configured by going into the settings of the paired Vitrum device in the Homey's Zones and Devices overview. There's 3 association groups for my 3 button switch. One association group for each button. The first group is for the first button, the second group for the second button, and the third group for the third button. The idea here is that you can specify the Z-wave node numbers of the devices you want each of these buttons/groups to send a signal when you switch a button on or off. The Homey is at number 1 so that's the number I put for all 3 groups/buttons. This should work. And it does; the signal does arrive at the Homey. 

But unfortunately, it is the 'wrong' type of signal, apparently. And that's the issue. It's not that the signal doesn't get to the Homey, it's that the software in the Homey decides that it shouldn't respond to this as an on/off signal. So, hopefully, with an app, that can 'read' the signals, this issue can be solved. The app could form the bridge between the 'wrong' signal and a simulated 'right' signal. I hope.

**Comment by novaflash** [comment]
tl;dr version:

Inclusion/exclusion is to add/remove a device to/from a Z-wave network.
Association is to connect devices in a Z-wave network to one another.
